backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -949,29 +970,35 @@ public ProjectPaginationDTO findProjects(
           projectPaginationDTO.setTotalRecords(0L);
           return projectPaginationDTO;
         }
-        List<String> orgIds =
+        List<String> orgWorkspaceIds =
             roleService.listMyOrganizations().stream()
-                .map(Organization::getId)
+                .map(Organization::getWorkspaceId)
                 .collect(Collectors.toList());
-        if (!orgIds.isEmpty()) {
-          finalPredicatesList.add(
-              builder.not(
-                  builder.and(
-                      projectRoot.get(ModelDBConstants.WORKSPACE).in(orgIds),
-                      builder.equal(
-                          projectRoot.get(ModelDBConstants.WORKSPACE_TYPE),
-                          WorkspaceType.ORGANIZATION_VALUE))));
+
+        List<GetResourcesResponseItem> items =
+            roleService.getResourceItems(accessibleProjectIds, ModelDBServiceResourceTypes.PROJECT);
+        for (GetResourcesResponseItem item : items) {
+          if (orgWorkspaceIds.contains(String.valueOf(item.getWorkspaceId()))) {
+            accessibleProjectIds.remove(item.getResourceId());
+          }
```

Comment:
What is this doing? I'm confused.

Reply:
Here we can not use the workspace name with UAC call because we have multiple org_workspace and if i used that call with  workspace then it will call UAC in for loop for each org_worksapce. So here i have call UAC once without org_worksapce and applying logic to remove those ids.